### PR TITLE
Make sure linting disallow unsupported json keys

### LINKED ISSUE #523: Make sure only valid keys are present in each entry
#522 found out that `note` was accepted, but didn't show up correctly, so we should make sure the linter gets these occurences and doesn't allow those to be merged.

### PR DESCRIPTION
**Pull request by tupaschoal**
Closes #523 

Funny enough it already paid off as each of the methods found an error in our JSON 🚀

### CODE REVIEW

File: script/validate_json.rb [hunk 1]
```diff
@@ -37,6 +40,23 @@ def get_transformed_name(site_object)
...
 
+def validate_accepted_keys(key)
+    key.keys.each do |entry_key|
+        if entry_key.start_with?('url_') || entry_key.start_with?('notes_')
+            # These have their own validation methods
+            next
+        end
+
+        unless SupportedEntryKeys.include?(entry_key)
+            STDERR.puts "Entry '#{key['name']}' has unsupported field:"\
```
Comment (Nightfirecat) [inline script/validate_json.rb:51]:
```suggestion
            STDERR.puts "Entry '#{key['name']}' has unsupported field: "\
```

**Comment by tupaschoal** [comment]
> Looks largely fine to me. Does this PR not add a check for `url` fields with unknown languages?

We already had `validate_localized_urls`

**Comment by PyroSamurai** [comment]
honestly, I thought we already had such linting in place to begin with otherwise I would have suggested it before our users did :/

**Comment by tupaschoal** [comment]
It's alright, now we have and everything will be fine :D

**Pull request merged**